api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPractitionerServiceImpl.java

```diff
@@ -92,17 +97,36 @@ public Practitioner create(@Nonnull Practitioner newResource) {
 	@Override
 	@Transactional(readOnly = true)
 	public IBundleProvider searchForPractitioners(PractitionerSearchParams practitionerSearchParams) {
-		IBundleProvider providerBundle = searchQuery.getQueryResults(practitionerSearchParams.toSearchParameterMap(), dao,
-		    translator, searchQueryInclude);
-		SearchParameterMap theParams = new SearchParameterMap();
-		IBundleProvider userBundle = userService.searchForUsers(theParams);
 		
-		if (!providerBundle.isEmpty() && !userBundle.isEmpty()) {
+		IBundleProvider providerBundle = null;
+		IBundleProvider userBundle = null;
+		
+		if (shouldSearchExplicitlyFor(practitionerSearchParams.getTag(), "provider")) {
+			providerBundle = searchQuery.getQueryResults(practitionerSearchParams.toSearchParameterMap(), dao, translator,
+			    searchQueryInclude);
+		}
+		
+		if (shouldSearchExplicitlyFor(practitionerSearchParams.getTag(), "user")) {
+			SearchParameterMap theParams = new SearchParameterMap();
+			userBundle = userService.searchForUsers(theParams);
+		}
+		
+		if (providerBundle != null && userBundle != null) {
 			return new TwoSearchQueryBundleProvider(providerBundle, userBundle, globalPropertyService);
-		} else if (providerBundle.isEmpty() && !userBundle.isEmpty()) {
+		} else if (providerBundle == null && userBundle != null) {
 			return userBundle;
 		}
 		
 		return providerBundle;
 	}
+	
+	protected boolean shouldSearchExplicitlyFor(TokenAndListParam tokenAndListParam, @Nonnull String valueToCheck) {
+		if (tokenAndListParam == null || tokenAndListParam.size() == 0 || valueToCheck.isEmpty()) {
+			return true;
+		}
+		
+		return tokenAndListParam.getValuesAsQueryTokens().stream().anyMatch(
+		    tokenOrListParam -> tokenOrListParam.doesCodingListMatch(Collections
+		            .singletonList(new InternalCodingDt(FhirConstants.OPENMRS_FHIR_EXT_PRACTITIONER_TAG, valueToCheck))));
```

Comment:
So I copied this method bring much verbatim from a similar method in the encounter service, but, honestly, I don't exactly understand what the this internal coding and this new constant I defined does... we *don't* include this coding in our actual request, but it still seems to match like it should. @ibacher ?

Reply:
The thing you copied and pasted from was intended to support this:

```
/Encounter?_tag=http://fhir.openmrs.org/ext/encounter-tag|encounter
```

So, in this case:

```
/Provider?_tag=http://fhir.openmrs.org/ext/provider-tag|provider
```

If this works, it's kind of by mistake or at least for reasons I don't fully understand... (FHIR Tags are coded elements so ideally they'd have both a system and a code).